cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataReaderContext.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.metadata;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The context for reading metadata from program
+ */
+public interface MetadataReaderContext {
+
+  /**
+   * Returns a set of {@link MetadataRecord} representing all metadata (including properties and tags) for the specified
+   * {@link MetadataEntity} in both {@link MetadataScope#USER} and {@link MetadataScope#SYSTEM}.
+   */
+  Set<MetadataRecord> getMetadata(MetadataEntity metadataEntity);
+
+  /**
+   * Returns a set of {@link MetadataRecord} representing all metadata (including properties and tags) for the specified
+   * {@link MetadataEntity} in the specified {@link MetadataScope}.
+   */
+  Set<MetadataRecord> getMetadata(MetadataScope scope, MetadataEntity metadataEntity);
```

Comment:
Why the result is a `Set`? Isn't that there should be just one `MetadataRecord` for a given entity in a given scope?